### PR TITLE
feat: passing rollupBeneficiary to sendDefiRollup(...)

### PR DESCRIPTION
**Pull request by benesjan**
# Description

When trying to test the Subsidy contract in an E2E test I stumbled upon a need to pass `rollupBeneficiary` to a rollup. This PR allows for that.

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [x] There are no unexpected formatting changes, or superfluous debug logs.
- [x] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewers next convenience.
- [x] NatSpec documentation of all the non-test functions is present and is complete.

### CODE REVIEW

**Comment by LHerskind** [comment]
I think that the `rollupBeneficiary` will only rarely be used in tests, so instead of bloating the sendDefiRollup, it might be better just to add a setter and getter of the value to store it inside the `BridgeTestBase`. Then there will only be changes for the test requiring the beneficiary.

**Comment by benesjan** [comment]
> I think that the `rollupBeneficiary` will only rarely be used in tests, so instead of bloating the sendDefiRollup, it might be better just to add a setter and getter of the value to store it inside the `BridgeTestBase`. Then there will only be changes for the test requiring the beneficiary.

Addressed in b085d312cc839b72b9a04246196caf2c606662c9

File: src/test/aztec/base/BridgeTestBase.t.sol [hunk 1]
```diff
@@ -8,6 +8,7 @@ import {BridgeTestBase} from "./BridgeTestBase.sol";
 contract BridgeTestBaseTest is BridgeTestBase {
     uint256 private constant VIRTUAL_ASSET_ID_FLAG_SHIFT = 29;
     uint256 private constant VIRTUAL_ASSET_ID_FLAG = 0x20000000; // 2 ** 29
+    uint256 internal constant ADDRESS_MASK = 0x00ffffffffffffffffffffffffffffffffffffffff;
```
Comment (LHerskind) [inline src/test/aztec/base/BridgeTestBase.t.sol:11]: Why does the mask include `00`?

File: src/test/aztec/base/BridgeTestBase.sol [hunk 2]
```diff
@@ -337,178 +349,178 @@ abstract contract BridgeTestBase is Test {
...
+            mstore(add(data, 0x1080), 0x0000000000000000000000000000000000000000000000000000000000000000)
+            mstore(add(data, 0x10a0), 0x0000000000000000000000000000000000000000000000000000000000000000)
+            mstore(add(data, 0x10c0), 0x0000000000000000000000000000000000000000000000000000000000000000)
+            mstore(add(data, 0x10e0), 0x0000000000000000000000000000000000000000000000000000000000000000)
+            mstore(add(data, 0x1100), 0x0000000000000000000000000000000000000000000000000000000000000000)
+            mstore(add(data, 0x1120), 0x0000000000000000000000000000000000000000000000000000000000000000)
+            mstore(add(data, 0x1140), 0x0000000000000000000000000000000000000000000000000000000000000000)
+            mstore(add(data, 0x1160), 0x0000000000000000000000000000000000000000000000000000000000000000)
+            mstore(add(data, 0x1180), 0x14e0f351ade4ba10438e9b15f66ab2e6389eea5ae870d6e8b2df1418b2e6fd5b)
+            mstore(add(data, 0x11a0), sload(rollupBeneficiary.slot))
```
Comment (LHerskind) [inline src/test/aztec/base/BridgeTestBase.sol:514]: Please use the same structure as the other storage values. e.g., put them up top so it is easy to see what is changed.

**Review by LHerskind — COMMENTED** [review]
Minor comments or following style. Good to go when these are fixed.

File: src/test/aztec/base/BridgeTestBase.sol (hunk 2 above)
Comment (benesjan) [reply src/test/aztec/base/BridgeTestBase.sol]: Addressed in a3eb8bce7dd21f52b8de828256e5d19ca167d0a8

File: src/test/aztec/base/BridgeTestBase.t.sol (hunk 1 above)
Comment (benesjan) [reply src/test/aztec/base/BridgeTestBase.t.sol]: Addressed in a3eb8bce7dd21f52b8de828256e5d19ca167d0a8

**Review by LHerskind — APPROVED** [review]

**Pull request merged**